enrichments/trace/internal/elastic/span.go

```diff
@@ -187,10 +187,17 @@ func (s *spanEnrichmentContext) Enrich(span ptrace.Span, cfg config.Config) {
 }
 
 func (s *spanEnrichmentContext) enrich(span ptrace.Span, cfg config.Config) {
+
+	// In OTel, a local root span can represent an outgoing call or a producer span.
+	// In such cases, the span is still mapped into a transaction, but enriched
+	// with additional attributes that are specific to the outgoing call or producer span.
+	isExitRootSpan := s.isTransaction && span.Kind() == ptrace.SpanKindClient || span.Kind() == ptrace.SpanKindProducer
```

Comment:
This check is still up for discussion: https://github.com/elastic/opentelemetry-lib/pull/125#discussion_r1863663594